fenix/app/src/main/java/org/mozilla/fenix/browser/TranslationsBinding.kt

```diff
@@ -68,13 +68,17 @@ class TranslationsBinding(
 
                 // Session Translations State Behavior (Tab)
                 val sessionTranslationsState = state.sessionState.translationsState
-                if (sessionTranslationsState.isTranslated) {
-                    val fromSelected = sessionTranslationsState.translationEngineState?.initialFromLanguage(
-                        translateFromLanguages,
-                    )
-                    val toSelected = sessionTranslationsState.translationEngineState?.initialToLanguage(
-                        translateToLanguages,
-                    )
+                val isEngineSupported = browserTranslationsState.isEngineSupported == true
+
+                if (sessionTranslationsState.isTranslated && isEngineSupported) {
```

Comment:
If the `isEngineSupported` on these individual items can be removed, I would recommend removing them to simplify the logic. 

(I think in this scenario, the values just go to null and/or don't change.)

I tested entering the browser menu with `browser.translations.simulateUnsupportedEngine` = true in `about:settings`. This logic nulled it out as expected. Might be worth testing if the `isEngineSupported` are really required.